test_runner/src/main/kotlin/ftl/args/yml/YamlDeviceFix.kt

```diff
@@ -0,0 +1,37 @@
+package ftl.args.yml
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ObjectNode
+import ftl.args.ArgsHelper.yamlMapper
+import ftl.run.exception.FlankGeneralError
+import ftl.util.loadFile
+import java.nio.file.Path
+
+fun fixDevices(yamlPath: Path) =
+    if (yamlPath.toFile().exists().not()) throw FlankGeneralError("Flank yml doesn't exist at path $yamlPath")
+    else yamlPath.loadYml().fixDevices().saveFixedYml(yamlPath)
+
+private fun Path.loadYml() = yamlMapper.readTree(loadFile(this))
+
+private fun JsonNode.fixDevices() = also {
+    getDevicesNode()?.getNotValidDevices()?.forEach { device ->
+        device.fixDeviceVersion()
+    }
+}
+
+internal fun JsonNode.getDevicesNode() = get(GCLOUD_NODE).get(DEVICES_NODE)
```

Comment:
I am thinking about changing extension functions to extension properties. As stated 
 in https://kotlinlang.org/docs/reference/coding-conventions.html#functions-vs-properties
rewriting `getDevicesNode()`, `getNotValidDevices()` etc into properties (example) `JsonNode.devicesNode` would result in  
`devicesNode.notValidDevices.filter { ...}`
We are operating on properties anyway so that always returns the same value. It's not crucial for your PR, I'd like to get you opinion @adamfilipow92 @piotradamczyk5

Reply:
As it is a getter, ofc It could be converted to property. But is a cosmetic change.

Reply:
Thanks, for this topic! 👍  On the first look, it's looking great. It can cache results so we can avoid computation the same things and in cases when we have ```getSomething()``` it fits better in my opinion.

Reply:
> As it is a getter, ofc It could be converted to property. But it's a cosmetic change.

Never said otherwise